Return-Path: mukawa@igel.co.jp
Received: from zmta01.collab.prod.int.phx2.redhat.com (LHLO
 zmta01.collab.prod.int.phx2.redhat.com) (10.5.81.8) by
 zmail09.collab.prod.int.phx2.redhat.com with LMTP; Tue, 26 Apr 2016
 01:39:41 -0400 (EDT)
Received: from int-mx11.intmail.prod.int.phx2.redhat.com (int-mx11.intmail.prod.int.phx2.redhat.com [10.5.11.24])
	by zmta01.collab.prod.int.phx2.redhat.com (Postfix) with ESMTP id 0E056185E36
	for <pmatilai@mail.corp.redhat.com>; Tue, 26 Apr 2016 01:39:41 -0400 (EDT)
Received: from mx1.redhat.com (ext-mx07.extmail.prod.ext.phx2.redhat.com [10.5.110.31])
	by int-mx11.intmail.prod.int.phx2.redhat.com (8.14.4/8.14.4) with ESMTP id u3Q5dexj001739
	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-GCM-SHA384 bits=256 verify=NO)
	for <pmatilai@redhat.com>; Tue, 26 Apr 2016 01:39:41 -0400
Received: from mail-pa0-f52.google.com (mail-pa0-f52.google.com [209.85.220.52])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by mx1.redhat.com (Postfix) with ESMTPS id 5FD14C049D4F
	for <pmatilai@redhat.com>; Tue, 26 Apr 2016 05:39:39 +0000 (UTC)
Received: by mail-pa0-f52.google.com with SMTP id bt5so2234157pac.3
        for <pmatilai@redhat.com>; Mon, 25 Apr 2016 22:39:39 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=igel-co-jp.20150623.gappssmtp.com; s=20150623;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=MSJ3uh/eHWdSz39+CLsgqwXBa7PdXoUeHF+aDBNUoQM=;
        b=lBx+N18Tl+nwQ55xGCDbxrR5567MwpY6VMI5yNXdgVLTRRbpnz3HPxvrSIuRGTzRnb
         i1cUEKUaUeDM9w/lSkGKEktnb5R8siw42/8ar+LXUW8Gylj1q7yQBKRYTL7QF6w8AD/E
         4hZu59whOYYXNFf5b3r2nzr+lDL/tFeeXkti/tF2rdRUzJ+nsZpAiU64bAOaEy6c+ppY
         V919oyvtCBKsJTw9qWdplykJzPhyvV9jgX+im9DQjRsm2SnKwpB3wU8T2PpipGI78gFF
         cHpkSWHFOqpucEnuqPWOBO8bXnYZlcVinEjIuUcOEx4fHSXfRRvmY+3IujrMzaHW7Ya0
         B6BA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20130820;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=MSJ3uh/eHWdSz39+CLsgqwXBa7PdXoUeHF+aDBNUoQM=;
        b=R0ScE8l83uR9rFnrCgtQ2WupzPUEMAfChLzWO6SiCv8CVbuwXTWzJiEnMZ5gIQWNRK
         oR5qJWjTutje8qJzf6PJc9wujjz43P5bmd2FmKkvLh5C25mb+5Wur4Gr7XUJFQZ1RXbT
         SuOGiST7adTYSKy51pw4qy6mmmUfa5HjEjzsLJ7cITW8IS4GthGbV0YZPPtF+6DcrhOv
         yhoWYQzYMDmCduNvmKI8blCcjGeSCAZdFnqaMjV1pVPDdmkSFConnlRvSSYSWVv0BJ9d
         wShypOM2pCUNeQ/T8KuhLqu80Pz0vieOFF54fqVxgD/XoUcwHjrtaRyvqRzPbq/Qx6oP
         teNw==
X-Gm-Message-State: AOPr4FWhwfN4OhrlPzt1GgzYwNYD72qYX4XKaKwIUlXv/8whN7gnBOy7Xu1HmhtHHtlHow==
X-Received: by 10.66.121.197 with SMTP id lm5mr881644pab.143.1461649178950;
        Mon, 25 Apr 2016 22:39:38 -0700 (PDT)
Received: from localhost.localdomain (napt.igel.co.jp. [219.106.231.132])
        by smtp.gmail.com with ESMTPSA id 79sm33137907pfq.65.2016.04.25.22.39.37
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-SHA bits=128/128);
        Mon, 25 Apr 2016 22:39:38 -0700 (PDT)
From: Tetsuya Mukawa <mukawa@igel.co.jp>
To: dev@dpdk.org
Cc: pmatilai@redhat.com, yuanhan.liu@linux.intel.com, huawei.xie@intel.com,
        Tetsuya Mukawa <mukawa@igel.co.jp>
Subject: [PATCH v2] vhost: Fix linkage of vhost PMD
Date: Tue, 26 Apr 2016 14:39:29 +0900
Message-Id: <1461649169-13967-1-git-send-email-mukawa@igel.co.jp>
In-Reply-To: <571DDDD8.6000000@igel.co.jp>
References: <571DDDD8.6000000@igel.co.jp>
X-RedHat-Spam-Score: -0.321  (BAYES_50,DCC_REPUT_00_12,DKIM_SIGNED,DKIM_VALID,RCVD_IN_DNSWL_LOW,RCVD_IN_MSPIKE_H3,RCVD_IN_MSPIKE_WL,SPF_PASS) 209.85.220.52 mail-pa0-f52.google.com 209.85.220.52 mail-pa0-f52.google.com <mukawa@igel.co.jp>
X-Scanned-By: MIMEDefang 2.68 on 10.5.11.24
X-Scanned-By: MIMEDefang 2.78 on 10.5.110.31

Currently, vhost PMD doesn't have linkage for librte_vhost, even though
it depends on librte_vhost APIs. This causes a linkage error if below
conditions are fulfilled.

 - DPDK libraries are compiled as shared libraries.
 - DPDK application doesn't link librte_vhost.
 - Above application tries to link vhost PMD using '-d' DPDK option.

The patch adds linkage for librte_vhost to vhost PMD not to cause an
above error.

Signed-off-by: Tetsuya Mukawa <mukawa@igel.co.jp>
Acked-by: Panu Matilainen <pmatilai@redhat.com>
---
 drivers/net/vhost/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/vhost/Makefile b/drivers/net/vhost/Makefile
index f49a69b..30b91a0 100644
--- a/drivers/net/vhost/Makefile
+++ b/drivers/net/vhost/Makefile
@@ -38,6 +38,7 @@ LIB = librte_pmd_vhost.a
 
 CFLAGS += -O3
 CFLAGS += $(WERROR_FLAGS)
+LDLIBS += -lrte_vhost
 
 EXPORT_MAP := rte_pmd_vhost_version.map
 
-- 
2.5.0

